io/src/lib.rs

```diff
@@ -52,6 +52,19 @@ pub trait Read {
     fn take(&mut self, limit: u64) -> Take<Self> { Take { reader: self, remaining: limit } }
 }
 
+/// A trait describing an input stream that uses an internal buffer when reading.
+pub trait BufRead: Read {
+    /// Returns data read from this reader, filling the internal buffer if needed.
+    fn fill_buf(&mut self) -> Result<&[u8]>;
+
+    /// Consumes data from the internal buffer.
```

Comment:
Since you have to modify the PR anyway, I suggest changing this to "Marks the buffered data up to `amount` as consumed" since the method actually doesn't consume anything.

Reply:
Used as suggested.